File: app/src/main/java/io/githup/limvot/mangaapp/Chapter.java
package io.githup.limvot.mangaapp;

import org.luaj.vm2.LuaTable;

/**
 * Created by nathan on 8/22/14.
 */
public class Chapter {
    private Manga parentManga;
    private LuaTable backingTable;
    private int chapterNum;
    public Chapter(Manga parent, LuaTable tableIn, int num) {
        parentManga = parent;
        backingTable = tableIn;
        chapterNum = num;
    }
    public String toString() {
        return backingTable.get("title").toString();
    }
    public LuaTable getTable() {
        return backingTable;
    }
    public int getNum() { return chapterNum; }
    public Manga getParentManga() {
        return parentManga;
    }
}

File: app/src/main/java/io/githup/limvot/mangaapp/Chapter.scala
package io.githup.limvot.mangaapp;

import org.luaj.vm2.LuaTable;

class Chapter(parentManga:Manga , backingTable:LuaTable, chapterNum:Int){

    override def toString(): String = getTitle()
    def getTitle(): String = backingTable.get("title").toString()

    def getTable() : LuaTable = backingTable

    def getNum() : Int = chapterNum
    
    def getParentManga() : Manga = parentManga

    override def equals(other:Any) : Boolean = {
        // One shudders at the horror that this is
        toString().equals(other.toString())
    }
}
